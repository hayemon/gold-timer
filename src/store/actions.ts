export { addReward, incrementTotalOverdrives } from './slices/statSlice'
export {
  setInactive,
  setLastValue,
  setOverdriveCount,
  setPressCount,
  setTimerCount,
} from './slices/timerSlice'
