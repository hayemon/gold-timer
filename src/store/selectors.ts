export { selectStat } from './slices/statSlice'
export { selectTimer } from './slices/timerSlice'
