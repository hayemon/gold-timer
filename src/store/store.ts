import {
  combineReducers,
  configureStore,
  PreloadedState,
} from '@reduxjs/toolkit'
import statReducer from './slices/statSlice'
import timerReducer from './slices/timerSlice'

const rootReducer = combineReducers({
  timer: timerReducer,
  stat: statReducer,
})

export const createStore = (preloadedState?: PreloadedState<RootState>) =>
  configureStore({
    reducer: rootReducer,
    preloadedState,
  })

export type RootState = ReturnType<typeof rootReducer>
export type AppStore = ReturnType<typeof createStore>
export type AppDispatch = AppStore['dispatch']
