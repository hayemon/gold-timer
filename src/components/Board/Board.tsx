import { Indicator, TreasureBox } from 'components'
import React from 'react'
import { selectTimer } from 'store/selectors'

const Board: React.FC = () => {
  const { inactive } = selectTimer()

  let className = 'treasure-box'
  if (inactive) {
    className += ' treasure-box__shaded'
  }

  return (
    <div className='board-container'>
      <TreasureBox />
      <Indicator />
    </div>
  )
}

export default Board
