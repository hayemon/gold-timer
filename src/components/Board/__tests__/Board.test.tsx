import { renderWithProviders } from 'store/testUtils'
import Board from '../Board'

describe('Board', () => {
  it('renders', () => {
    const { asFragment } = renderWithProviders(<Board />)
    expect(asFragment()).toMatchSnapshot()
  })
})
