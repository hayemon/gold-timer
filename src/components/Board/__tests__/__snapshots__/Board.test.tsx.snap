// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders 1`] = `
<DocumentFragment>
  <div
    class="board-container"
  >
    <div
      class="treasure-box-container"
    >
      <h1
        class="ant-typography"
      >
        0 Coins!!!
      </h1>
      <button
        class="ant-btn"
        style="width: 200px; height: 200px; margin: 0px; padding: 0px;"
        type="button"
      >
        <div
          class="treasure-box"
        />
      </button>
    </div>
    <div
      class="indicator-container"
    />
  </div>
</DocumentFragment>
`;
